### PR TITLE
chore: Update Actions labeler and labeler config to v5.0

### PR DESCRIPTION
**Pull request by umbr4g3**
Closes: #7008 

## What is the purpose of the change

This pull request updates the Actions labeler version to v5.0 and updates the `labeler.yml` file to be compatible with the v5.0 config file structure.

The new config file matches the existing functionality, and in the future it can be updated to take advantage of the new features from the v5.0 actions/labeler.

## Testing and Verifying

This change added tests and can be verified as follows:

* Since this workflow runs on `pull_request_target` it needs to be tested with a pull request to a branch with the updated workflow.
* Thankfully, it was possible to manually test this with within my fork, and I ran through a few cases here: https://github.com/umbr4g3/osmosis/pull/1/files
    * I modified specific files within my PR to trigger the action, and it added the expected labels.
* Additionally, reviewers can look through https://github.com/actions/labeler?tab=readme-ov-file#basic-examples to confirm that the changes align with the expected structure and match the behavior of the old file.

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [ ] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [x] N/A

### CODE REVIEW

**Review by p0mvn — APPROVED** [review]
Amazing, ty!

File: CHANGELOG.md [hunk 1]
```diff
@@ -79,7 +79,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * [#7068](https://github.com/osmosis-labs/osmosis/pull/7068) Add CI action that check for typo on new PRs (Thanks @codespell)
 * [#7093](https://github.com/osmosis-labs/osmosis/pull/7093),[#7100](https://github.com/osmosis-labs/osmosis/pull/7100) Lower CPU overheads of the Osmosis epoch.
 * [#7106](https://github.com/osmosis-labs/osmosis/pull/7106) Halve the time of log2 calculation (speeds up TWAP code)
-
+* [#7134](https://github.com/osmosis-labs/osmosis/pull/7134) Update Actions labeler and labeler config to v5.0
```
Comment (p0mvn) [inline CHANGELOG.md:82]: TBH, we don't need a changelog for this. There is a "no changelog" label that can be applied to make the CI happy

Comment (umbr4g3) [reply CHANGELOG.md]: Ah, i reverted it.

**Comment by p0mvn** [comment]
Thank you for the change @umbr4g3

**Pull request merged**